### PR TITLE
Attacking resin walls channels the next input.

### PR DESCRIPTION
**Pull request by Losenis**
## About The Pull Request
Attacks against resin walls now automatically channel the next attack. No balance changes.
Code mostly by RikuTheKiller. All credit and shitflinging should go to him.

## Why It's Good For The Game
Less spamclicking, less arthritis.

https://github.com/tgstation/TerraGov-Marine-Corps/assets/72712909/b9614d7a-2993-4707-b08b-31c46537a0c4

## Changelog

:cl:
qol: Attacking a resin wall automatically channels the next attack.
/:cl:

### CODE REVIEW

**Comment by Losenis** [comment]
TMable while Riku works on expanding it to most things, do not merge

**Comment by isotope436** [comment]
this is a PR i can get behind

**Comment by Losenis** [comment]
Wait oh fuck wrong button undrafted fuck

**Comment by FosterGrime-Heymynameslouie** [comment]
holy shit

**Comment by Xander3359** [comment]
The 4 month comeback is so real.

File: code/game/turfs/walls/resin.dm [hunk 1]
```diff
@@ -129,10 +132,19 @@
...
 	take_damage(damage, BRUTE, MELEE)
 	playsound(src, "alien_resin_break", 25)
 
+	if(previous_type == type && do_after(user, I.attack_speed, TRUE, src, BUSY_ICON_GENERIC))
+		attackby(I, user, params)
+
+/turf/closed/wall/resin/CanAllowThrough(atom/movable/mover, turf/target)
+	. = ..()
+	if(istype(mover) && CHECK_BITFIELD(mover.flags_pass, PASSGLASS))
+		return !opacity
```
Comment (TiviPlus) [inline code/game/turfs/walls/resin.dm:146]: Why are you making it so people can go through walls

**Review by TiviPlus — COMMENTED** [review]
rest ok i guess

**Pull request closed without merging**